mozregression/launchers.py

```diff
@@ -388,6 +388,72 @@ def get_app_info(self):
         return self.app_info
 
 
+@REGISTRY.register('gve')
+class GeckoViewExampleLauncher(Launcher):
+    app_info = None
+    adb = None
+    package_name = None
+    profile_class = FirefoxRegressionProfile
+    remote_profile = None
+
+    @classmethod
+    def check_is_runnable(cls):
+        try:
+            devices = ADBHost().devices()
+        except ADBError as adb_error:
+            raise LauncherNotRunnable(str(adb_error))
+        if not devices:
+            raise LauncherNotRunnable("No android device connected."
+                                      " Connect a device and try again.")
+
+    def _install(self, dest):
+        # get info now, as dest may be removed
+        self.app_info = safe_get_version(binary=dest)
+        self.package_name = self.app_info.get("package_name",
+                                              "org.mozilla.geckoview_example")
+        self.adb = ADBAndroid(require_root=False)
+        try:
+            self.adb.uninstall_app(self.package_name)
+        except ADBError, msg:
+            LOG.warning(
+                "Failed to uninstall %s (%s)\nThis is normal if it is the"
+                " first time the application is installed."
+                % (self.package_name, msg)
+            )
+        self.adb.install_app(dest)
+
+    def _start(self, profile=None, addons=(), cmdargs=(), preferences=None):
+        # we don't handle addons on the profile
+        profile = self._create_profile(profile=profile,
+                                       preferences=preferences)
+        # send the profile on the device
+        self.remote_profile = "/".join([self.adb.test_root,
+                                       os.path.basename(profile.profile)])
+        if self.adb.exists(self.remote_profile):
+            self.adb.rm(self.remote_profile, recursive=True)
+        LOG.debug("Pushing profile to device (%s -> %s)" % (
+            profile.profile, self.remote_profile))
+        self.adb.push(profile.profile, self.remote_profile)
+
+        LOG.debug("Launching geckoview_example")
+        self.adb.launch_activity(self.package_name,
+                                 activity_name="GeckoViewActivity",
+                                 extra_args=["-profile", self.remote_profile],
+                                 e10s=True)
+
+    def _wait(self):
+        while self.adb.process_exist(self.package_name):
+            time.sleep(0.1)
+
+    def _stop(self):
+        self.adb.stop_application(self.package_name)
+        if self.adb.exists(self.remote_profile):
+            self.adb.rm(self.remote_profile, recursive=True)
+
+    def get_app_info(self):
+        return self.app_info
+
+
```

Comment:
It looks like this is 99% the same as the FennecLauncher class elsewhere in the file. Could you create a base class called `AndroidLauncher` and have both this and FennecLauncher inherit from it?

Reply:
Yup, will do